sklearn_numba_dpex/common/kernels.py

```diff
@@ -728,13 +707,13 @@ def partial_sum_reduction(
         # local memory of size (n_sub_groups_per_work_group, sub_group_size) (i.e one
         # slot for each work item and two items in the window), and yet again
         # contiguous work items write into contiguous slots.
-        local_values = dpex.local.array(local_values_size, dtype=dtype)
+        local_values = dpex.local.array(work_group_shape, dtype=dtype)
 
         # The current work item use the following second coordinate (given by the
         # position of the window in the grid of windows, and by the local position of
         # the work item in the 2D index):
         col_idx = (
-            (group_id // n_blocks_per_col) * sub_group_size + local_col_idx
+            (dpex.get_group_id(one_idx) * sub_group_size) + local_col_idx
             )
```

Comment:
black:

```suggestion
        )
```

which makes me think that I wanted to open a PR to fix the black problem but completely forgot about it. I will wait for this PR to me finalized first though.